lib/std/math/log_int.zig

```diff
@@ -0,0 +1,79 @@
+const std = @import("../std.zig");
+const math = std.math;
+const testing = std.testing;
+const assert = std.debug.assert;
+const Log2Int = math.Log2Int;
+
+/// Returns the logarithm of `x` for the provided `base`, rounding down to the nearest integer.
+/// Asserts that `base > 1` and `x != 0`.
+pub fn log_int(comptime T: type, base: T, x: T) Log2Int(T) {
+    if (@typeInfo(T) != .Int or @typeInfo(T).Int.signedness != .unsigned)
+        @compileError("log_int requires an unsigned integer, found " ++ @typeName(T));
+
+    assert(base > 1 and x != 0);
+
+    var exponent: Log2Int(T) = 0;
+    var power: T = 1;
+    while (power <= x / base) {
+        power *= base;
+        exponent += 1;
+    }
+
+    return exponent;
+}
+
+test "log" {
```

Comment:
This is missing the case for 0 and 1 bits for consistency.

Reply:
Since `base` must be bigger than 1 we need at least 2 bits. I don't see what we could test with 0 or 1 bits as `log_int` will always panic on the `assert(base > 1 and x != 0)`.